### DISCUSSION TITLE
Error when running spacy download into docker on Apple M1 chip

### DISCUSSION THREAD

**Discussion by marcelpsaraiva** [explosion/spaCy#10025] [Help: Installation] [answered]
I'm in a MacBook Pro with M1 Pro chip, and I've built a Docker to install Tensorflow, spacy, etc.
My Dockerfile:

`
FROM --platform=linux/x86_64 python:3.8
ENV PYTHONUNBUFFERED 1

RUN apt-get update && apt-get -y upgrade
RUN pip install --upgrade pip

RUN pip install TensorFlow==2.3.4
RUN pip install rasa==2.8.0
RUN pip install spacy 
RUN python -m spacy download pt_core_news_lg
`

I've added platform tag (linux/x86_64) in line 1, to install correctly the TensorFlow and spacy packages. But, when run the spacy download command, occurs this error.

` 
=> ERROR [13/13] RUN python -m spacy download pt_core_news_lg                                                                                                                                  2.3s
------
 > [13/13] RUN python -m spacy download pt_core_news_lg:
#17 2.290 2022-01-10 19:36:31.774553: F tensorflow/core/lib/monitoring/sampler.cc:42] Check failed: bucket_limits_[i] > bucket_limits_[i - 1] (0 vs. 10)
#17 2.290 qemu: uncaught target signal 6 (Aborted) - core dumped
#17 2.300 Aborted
------
executor failed running [/bin/sh -c python -m spacy download pt_core_news_lg]: exit code: 134
`

Any idea?

**Comment by adrianeboyd** [comment, marked as answer]
Does this work if you temporarily comment out the install steps for tensorflow and rasa? In the background, spacy tries to import tensorflow by default if it's installed, so if you only install spacy and then `python -m spacy` runs, then it's probably just related to the tensorflow install.

**Reply by marcelpsaraiva** [reply]
First of all, thank you for attention.

Well, I've commented the TensorFlow and rasa install command. And it's works partially, doesn't occur the problem related above, but the container is down, because doesn't find rasa.
And I've tested the docker file below:

`
FROM --platform=linux/x86_64 python:3.8
ENV PYTHONUNBUFFERED 1

RUN apt-get update && apt-get -y upgrade
RUN pip install --upgrade pip

RUN pip install rasa==2.8.0
RUN pip install spacy
RUN python -m spacy download pt_core_news_lg
`

As you see, I put off TensorFlow install, and maintain rasa install.
And the problem occurs.

=> ERROR [12/12] RUN python -m spacy download pt_core_news_lg                              3.1s 
------                                                                                           
 > [12/12] RUN python -m spacy download pt_core_news_lg:
#17 3.074 2022-01-11 13:42:37.228475: F tensorflow/core/lib/monitoring/sampler.cc:42] Check failed: bucket_limits_[i] > bucket_limits_[i - 1] (0 vs. 10)
#17 3.075 qemu: uncaught target signal 6 (Aborted) - core dumped
#17 3.091 Aborted
------
executor failed running [/bin/sh -c python -m spacy download pt_core_news_lg]: exit code: 134
ERROR: Service 'chatbot-rasa-backend-compose' failed to build : Build failed

**Reply by adrianeboyd** [reply]
Rasa probably requires tensorflow by default so it ends up installed anyway. Since it's looking like the problem isn't with spacy directly, a rasa forum might be a better place to ask.
